File: libs/core/persistence/lib/Models/Truck.cs
using Petrologistic.Core.Persistence.Lib.Models.Location;

namespace Petrologistic.Core.Persistence.Lib.Models;

public class Truck : BaseModel
{
  public string Name { get; set; } = string.Empty;
  public int Number { get; set; } = 0;
  public string Description { get; set; } = string.Empty;
  public string TrucksLicense { get; set; } = string.Empty;
  public string TrailersLicense { get; set; } = string.Empty;
  public string Permit { get; set; } = string.Empty;
  public string System { get; set; } = string.Empty;
  public GeoCoordinates Position { get; set; } = default!;
  public ICollection<Compartment> Compartments { get; set; } = default!;
}

File: libs/features/boards/frontend/src/lib/domain/truck.ts
import { Compartment } from "./compartment"

export interface Truck {
    id: string,
    name: string,
    description: string,
    number: number,
    trucksLicense: string,
    trailersLicense: string,
    permit: string,
    longitude: number,
    latitude: number
    system: string,
    selected: boolean,
    compartments: Compartment[]
}